src/py42/clients/detectionlists/departing_employee.py

```diff
@@ -72,53 +73,83 @@ def remove(self, user_id):
         data = {u"userId": user_id, u"tenantId": tenant_id}
         return self._session.post(uri, data=json.dumps(data))
 
-    def _get_departing_employees_page(
+    def get_page(
         self,
-        tenant_id=None,
         filter_type=None,
         sort_key=u"CREATED_AT",
         sort_direction=u"DESC",
-        page_num=None,
-        page_size=None,
+        page_num=1,
+        page_size=_PAGE_SIZE,
     ):
+        """Gets a single page of Departing Employees.
 
-        uri = self._uri_prefix.format(u"search")
-        data = {
-            u"tenantId": tenant_id,
-            u"pgSize": 100,
-            u"pgNum": page_num,
-            u"filterType": filter_type,
-            u"srtKey": sort_key,
-            u"srtDirection": sort_direction,
-        }
-        return self._session.post(uri, data=json.dumps(data))
+        Args:
+            filter_type (str, optional): Valid filter types. Defaults to None.
+            sort_key (str, optional): Sort results based by field. Defaults to "CREATED_AT".
+            sort_direction (str. optional): ``ASC`` or ``DESC``. Defaults to "DESC".
+            page_num (str or int, optional): The page number to request. Defaults to 1.
+            page_size (str or int, optional): The items to have per page. Defaults to 100.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+
+        return self._get_page(
+            tenant_id=self._user_context.get_current_tenant_id(),
+            filter_type=filter_type,
+            sort_key=sort_key,
+            sort_direction=sort_direction,
+            page_num=page_num,
+            page_size=page_size,
+        )
 
     def get_all(
         self, filter_type=u"OPEN", sort_key=u"CREATED_AT", sort_direction=u"DESC"
     ):
         """Gets all Departing Employees.
 
         Args:
-            filter_type (str, optional): Filter results by status. Defaults to "OPEN".
-            sort_key (str, optional): Key to sort results on. Options: (``CREATED_AT``,
-                ``DEPARTURE_DATE``, ``DISPLAY_NAME``, ``NUM_EVENTS``, ``TOTAL_BYTES``). Defaults to
-                ``CREATED_AT``.
-            sort_direction (str, optional): Sort direction. Options: (``ASC``, ``DESC``). Defaults
-                to ``DESC``.
+            filter_type (str, optional): Valid filter types. Defaults to None.
+            sort_key (str, optional): Sort results based by field. Defaults to "CREATED_AT".
+            sort_direction (str. optional): ``ASC`` or ``DESC``. Defaults to "DESC".
 
         Returns:
             generator: An object that iterates over :class:`py42.response.Py42Response` objects
             that each contain a page of departing employees.
         """
         return get_all_pages(
-            self._get_departing_employees_page,
+            self._get_page,
```

Comment:
you could use `self.get_page` here, and then you wouldn't have to include the code for gettting the tenant id.

Reply:
see @timabrmsn's note re: how extra calls to get the tenant id don't really incur a penalty